src/itype.h

```diff
@@ -240,7 +240,7 @@ struct islot_gun : common_firing_data {
     /**
      * What type of ammo this gun uses.
      */
-    std::string ammo;
+    itype_id ammo = "NULL";
```

Comment:
Ammo isn't an item type, it's an ammo type. For example item types could be "arrow_metal" or "arrow_wood", but both have the ammo type "arrow". Those ids refer to different things. `ammotype` is the correct type to use here.